Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-programming/questions-programming.html

```diff
@@ -4,62 +4,66 @@
       <h2>Question</h2>
       <h5>Programming</h5>
     </div>
-    <form>
-      <div class="form-fields">
+    <form #codeSnippetForm="ngForm" (ngSubmit)="onSubmit(codeSnippetForm)">
+      <div class="form-fields" >
         <div class="form-group">
           <label class="h4">Section</label>
-          <select class="form-control half-width">
-            <option>Logical Ability</option>
-            <option>Option 1</option>
-            <option>Option 2</option>
-            <option>Option 3</option>
+          <select class="form-control half-width" id="categorySelect" (change)="selectCategory($event.target.value)" ngModel required name="categorySelect" #categorySelect="ngModel">
+            <option selected></option>
+            <option *ngFor="let category of categoryList">{{category.categoryName}}</option>
           </select>
-          <div class="errors-container"></div>
+          <div *ngIf="categorySelect.errors && (categorySelect.dirty || categorySelect.touched)" class="errors-container">
+              <p class="text-danger">Please select a category</p>
+          </div>
         </div>
         <div class="form-group">
           <label class="h4">Difficulty Level</label>
-          <select class="form-control small-width">
-            <option>Easy</option>
+          <select class="form-control small-width" name="difficulty" (change)="selectDifficulty($event.target.value)">
+            <option selected>Easy</option>
             <option>Medium</option>
             <option>Hard</option>
           </select>
-          <div class="errors-container"></div>
+          <!--<div class="errors-container"></div>-->
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=Web%3AAvoidCommentedOutCodeCheck)